File: ReleaseNotes.Generator/Domain/RepositoryItemPath.cs
using System.ComponentModel.DataAnnotations;

namespace ReleaseNotes.Generator.Domain
{
    public class RepositoryItemPath : EntityBase<int>
    {
        [Required]
        public string Path { get; set; }

        [Required]
        public int BranchId { get; set; }

        public string LastCommitId { get; set; }

        public Branch Branch{ get; set; }
    }
}

File: ReleaseNotesGenerator/ClientApp/app/common/classes/RepositoryItemPath.ts

    import { Branch } from './Branch';
import { EntityBase } from './EntityBase';
    export class RepositoryItemPath extends EntityBase<number> {
        
        path: string;
        branchId: number;
        lastCommitId: string;
        lastCommitDateTime: Date;
        branch: Branch;
        }

    